lightning/src/ln/channel.rs

```diff
@@ -5941,6 +5941,23 @@ impl<Signer: Sign> Channel<Signer> {
 		self.update_time_counter += 1;
 		(monitor_update, dropped_outbound_htlcs)
 	}
+
+	pub fn inflight_htlc_sources(&self) -> Vec<&HTLCSource> {
+		let holding_cell_sources: Vec<&HTLCSource> = self.holding_cell_htlc_updates.iter()
+			.flat_map(|htlc_update| {
+				match htlc_update {
+					HTLCUpdateAwaitingACK::AddHTLC { source, .. } => { Some(source) }
+					_ => { None }
+				}
+			})
+			.collect();
+
+		let pending_outbound_sources: Vec<&HTLCSource> = self.pending_outbound_htlcs.iter()
+			.map(|htlc| &htlc.source)
+			.collect();
+
+		return holding_cell_sources.into_iter().chain(pending_outbound_sources.into_iter()).collect();
```

Comment:
Avoids vec allocations (to avoid heap fragmentation): 
```suggestion
		self.holding_cell_htlc_updates.iter()
			.flat_map(|htlc_update| {
				match htlc_update {
					HTLCUpdateAwaitingACK::AddHTLC { source, .. } => { Some(source) }
					_ => { None }
				}
			})
			.chain(self.pending_outbound_htlcs.iter().map(|htlc| &htlc.source)).collect()
```

Reply:
soo much nicer 😅